torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -409,12 +445,26 @@ ProcessGroupGloo::Options::Options()
 
 namespace {
 
+#ifdef _WIN32
+void init_winsock() {
+  WSADATA wsa_data;
+  int res;
+  res = WSAStartup(MAKEWORD(2, 2), &wsa_data);
+  if (res != 0) {
+    throw std::system_error(res, std::system_category());
+  }
+}
+#endif
+
 // Gloo assumes that this machine's hostname can always be resolved
 // to an address. If it doesn't it throws a runtime error saying
 // that it can't be resolved. Instead of catching it, we choose
 // to proactively check if an address can be resolved, so we can
 // gracefully fall back to an alternative if it doesn't.
 bool doesHostnameResolveToUsableAddress(const std::string& hostname) {
+#ifdef _WIN32
+  init_winsock();
+#endif
```

Comment:
Instead of guarding function call with `#ifdef _WIN32` can't we just add a dummy implementation of the function for non-windows platforms? (And call it something neutral, like `netInitialize`)